core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -872,6 +917,387 @@ export class ExplorationEditor extends BaseUser {
+        elements.map(async el => {
+          const textContent = await el.evaluate(node => node.textContent);
+          return textContent === property ? el : null;
+        })
+      );
+
+      const filteredElements = propertyElements.filter(el => el !== null);
+
+      if (filteredElements.length === 0) {
+        throw new Error(`Property ${property} not found`);
+      }
+
+      values[property] = (
+        await Promise.all(
+          filteredElements.map(async el => {
+            if (el) {
+              return el.evaluate(el => {
+                if (
+                  el.nextElementSibling &&
+                  el.nextElementSibling.nextElementSibling
+                ) {
+                  return el.nextElementSibling.nextElementSibling.textContent;
+                } else {
+                  throw new Error(
+                    `Unable to find third sibling for property ${property}`
+                  );
+                }
+              });
+            } else {
+              throw new Error(`Element is null for property ${property}`);
+            }
+          })
+        )
+      ).filter(value => value !== null) as string[];
+    }
+
+    for (let property in values) {
+      if (values[property][0] === values[property][1]) {
+        throw new Error(`No changes are reflected in ${property}`);
+      }
+      showMessage(`Changes are reflected in ${property}.`);
+    }
+  }
+
+  /**
+   * Verifies if metadata changes are reflected for each property within a single code block.
+   * @param {string[]} properties - The properties to check for.
+   */
+  async expectMetadataChangesInProperties(properties: string[]): Promise<void> {
+    await this.page.waitForSelector(viewMetadataChangesButton, {visible: true});
+    await this.clickOn(viewMetadataChangesButton);
+
+    await this.expectChangesInProperties(properties);
+
+    await this.clickOn(closeMetadataModal);
+    await this.page.waitForSelector(closeMetadataModal, {hidden: true});
+  }
+
+  /**
+   * Verifies if exploration state changes are reflected for each property within a single code block.
+   * @param {string[]} properties - The properties to check for.
+   */
+  async expectStateChangesInProperties(properties: string[]): Promise<void> {
+    // Add two spaces in front of each property to match the formatting in the UI.
+    const formattedProperties = properties.map(property => `  ${property}`);
+
+    await this.clickOn(testNodeBackground);
+    await this.expectChangesInProperties(formattedProperties);
+
+    await this.clickOn(closeStateModal);
+  }
+
+  /**
+   * Resets the comparison results by clicking on the reset graph button.
+   * @returns {Promise<void>}
+   */
+  async resetComparisonResults(): Promise<void> {
+    await this.page.waitForSelector(resetGraphButton);
+    await this.clickOn(resetGraphButton);
+  }
+
+  /**
+   * Finds the options button for a specific version.
+   * @param {number} version - The version number to find.
```

Comment:
Throughout the documentation and code of this file, the terms `revisionNumber` and `version` were used. Do both terms refer to the same concept? If so, then I prefer you stick with using the term `version` instead of `revisionNumber` within this file.

Reply:
done.

Reply:
done